tests/logging-utils/test_DEBUG2_logging.py

```diff
@@ -32,3 +33,47 @@ def test_extended_debug_logger(caplog, DEBUG2_enabled):
     assert record.levelno == 8
     assert record.args == (1,)
     assert record.message == "message 1"
+
+
+def test_caching_of_debug2_when_disabled(DEBUG2_installed):
+    # we need a unique logger because loggers are cached
+    logger = logging.getLogger("testing-{}".format(uuid.uuid4()))
+
+    assert logger.isEnabledFor(DEBUG2_LEVEL_NUM) is False
+
+    assert 'show_debug2' not in logger.__dict__
+    assert logger.show_debug2 is False
+    # cached property should have inserted it into the dict
+    assert 'show_debug2' in logger.__dict__
+
+    assert 'debug2' not in logger.__dict__
+    assert logger.debug2('this should actually call the function') is None
+    assert 'debug2' in logger.__dict__
+    assert logger.debug2('should not do anything but hit the lambda') is None
+
+    # now see that it always returns the value from `__dict__`
+    logger.__dict__['show_debug2'] = 100
+    assert logger.show_debug2 == 100
+
+
+def test_caching_of_debug2_when_enabled(DEBUG2_installed):
+    # we need a unique logger because loggers are cached
+    logger = logging.getLogger("testing-{}".format(uuid.uuid4()))
+
+    assert logger.isEnabledFor(DEBUG2_LEVEL_NUM) is False
+    logger.setLevel(DEBUG2_LEVEL_NUM)
+    assert logger.isEnabledFor(DEBUG2_LEVEL_NUM) is True
+
+    assert 'show_debug2' not in logger.__dict__
+    assert logger.show_debug2 is True
+    # cached property should have inserted it into the dict
+    assert 'show_debug2' in logger.__dict__
+
+    assert 'debug2' not in logger.__dict__
+    assert logger.debug2('this should actually call the function') is None
+    assert 'debug2' not in logger.__dict__
+    assert logger.debug2('this should still call the function') is None
```

Comment:
Any way to check that the log is actually being emitted?

Reply:
Yeah, gonna add a `caplog` test.